lib/VM/JSLib/TypedArray.cpp

```diff
@@ -1746,6 +1746,126 @@ typedArrayPrototypeToLocaleString(void *, Runtime &runtime, NativeArgs args) {
   return HermesValue::encodeStringValue(*builder->getStringPrimitive());
 }
 
+inline static bool isValidIntegerIndex(
+    Runtime &runtime,
+    Handle<JSTypedArrayBase> O,
+    const HermesValue &index) {
```

Comment:
Taking a `HermesValue` here is potentially error-prone. It looks like this is only ever called with a number - could we change this to just take `double` directly and avoid the potential of accidentally passing in an object pointer?

Reply:
My idea was to encapsulate the value and type checking. If this function would accept a double, 3 if conditions would need to be moved to caller making this a one-liner. Is there some alternative for this? 🤔